lib/linked_list.rb

```diff
@@ -7,39 +7,61 @@ def initialize
     @head = nil
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity O(1)
+  # Space complexity O(1)
   def add_first(data)
-
+    a = Node.new(data)
+    a.next = @head
+    @head = a
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(0)????
   def get_first
```

Comment:
What if `@head` is `nil`?

Reply:
OMG I did fixed all the bugs, and passed all the tests, but I forgot to push the last commit to git. Thank you for all the comments!